ffi/src/lib.rs

```diff
@@ -1968,10 +2012,309 @@ pub extern "C" fn create_otel_span(request_type: RequestType) -> u64 {
 pub extern "C" fn create_batch_otel_span() -> u64 {
     let command_name = "Batch";
 
+    // Create span and convert to pointer
     let span = GlideOpenTelemetry::new_span(command_name);
     let arc = Arc::new(span);
     let ptr = Arc::into_raw(arc);
-    ptr as u64
+    let span_ptr = ptr as u64;
+
+    logger_core::log_debug(
+        "ffi_otel",
+        format!(
+            "create_batch_otel_span: Successfully created batch span with pointer 0x{span_ptr:x}",
+        ),
+    );
+    span_ptr
+}
+
+/// Creates an OpenTelemetry batch span with a parent span and returns a pointer to the span as u64.
+/// This function creates a child span with the name "Batch" under the provided parent span.
+/// Returns 0 on failure.
+///
+/// # Parameters
+/// * `parent_span_ptr`: A u64 pointer to the parent span created by create_otel_span, create_named_otel_span, or create_batch_otel_span
+///
+/// # Returns
+/// * A u64 pointer to the created child batch span, or 0 if creation fails
+///
+/// # Safety
+/// * `parent_span_ptr` must be a valid pointer to a [`Arc<GlideSpan>`] span created by [`create_otel_span`], [`create_named_otel_span`], or [`create_batch_otel_span`], or 0.
+/// * If `parent_span_ptr` is 0 or invalid, the function will create an independent batch span as fallback.
+#[unsafe(no_mangle)]
+pub extern "C" fn create_batch_otel_span_with_parent(parent_span_ptr: u64) -> u64 {
```

Comment:
This needs to be marked `unsafe` because it has safety invariants that need to be upheld by the caller.